tests/e2e/raw_block_volume.go

```diff
@@ -596,4 +606,352 @@ var _ = ginkgo.Describe("raw block volume support", func() {
+			volumeID, pod.Spec.NodeName))
+		var vmUUID string
+		if vanillaCluster {
+			vmUUID = getNodeUUID(ctx, client, pod.Spec.NodeName)
+		} else if guestCluster {
+			vmUUID, err = getVMUUIDFromNodeName(pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			verifyCRDInSupervisorWithWait(ctx, f, pod.Spec.NodeName+"-"+svcPVCName,
+				crdCNSNodeVMAttachment, crdVersion, crdGroup, true)
+		}
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), fmt.Sprintf("Volume is not attached to the node, %s", vmUUID))
+
+		// Write and read some data on raw block volume inside the pod.
+		// Use same devicePath for raw block volume here as used inside podSpec by createPod().
+		// Refer setVolumes() for more information on naming of devicePath.
+		volumeIndex := 1
+		devicePath := fmt.Sprintf("%v%v", pod_devicePathPrefix, volumeIndex)
+		ginkgo.By(fmt.Sprintf("Write and read data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		verifyDataOnRawBlockVolume(namespace, pod.Name, devicePath, "This is write before expansion", 0)
+
+		defer func() {
+			// Delete Pod.
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client,
+				pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(),
+				fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+			if guestCluster {
+				ginkgo.By("Waiting for 30 seconds to allow CnsNodeVMAttachment controller to reconcile resource")
+				time.Sleep(waitTimeForCNSNodeVMAttachmentReconciler)
+				verifyCRDInSupervisorWithWait(ctx, f, pod.Spec.NodeName+"-"+svcPVCName,
+					crdCNSNodeVMAttachment, crdVersion, crdGroup, false)
+			}
+		}()
+
+		ginkgo.By("Increase PVC size and verify online volume resize")
+		increaseSizeOfPvcAttachedToPod(f, client, namespace, pvc, pod)
+
+		// Verify original data on raw block volume after expansion
+		// Write data on expanded space to verify the expansion is sucessful and accessible.
+		ginkgo.By(fmt.Sprintf("Verify previously written data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		readDataFromRawBlockVolume(namespace, pod.Name, devicePath, "This is write before expansion")
+		ginkgo.By(fmt.Sprintf("Write random data of total size %v on raw volume at path %v inside pod %v",
+			volsizeInMiBAfterExpansion, pod.Spec.Containers[0].VolumeDevices[0].DevicePath, pod.Name))
+		writeDataOnRawBlockVolume(namespace, pod.Name, devicePath, "", volsizeInMiBAfterExpansion)
```

Comment:
let's use a `verifyDataOnRawBlockVolume()` here

Reply:
done, since the size of the volume is large, writing few MBs on the expanded part and verifying it now.